src/main/java/com/iota/iri/network/replicator/ReplicatorSourceProcessor.java

```diff
@@ -152,11 +152,11 @@ public void run() {
                     for (int i=0; i<TRANSACTION_PACKET_SIZE; i++) {
                         crc32.update(data[i]);
                     }
-                    String crc32_string = Long.toHexString(crc32.getValue());
+                    StringBuilder crc32_string = new StringBuilder(Long.toHexString(crc32.getValue()));
                     while (crc32_string.length() < ReplicatorSinkProcessor.CRC32_BYTES) {
```

Comment:
Same here - just using StringUtils.leftPad(string,size,pad). Get rid of the concatenation altogether.